src/chunk_scan.c

```diff
@@ -252,9 +252,9 @@ ts_chunk_scan_by_constraints(const Hyperspace *hs, const List *dimension_vecs,
 				prev_chunk_oid = chunk->table_id;
 			}
 
+			MemoryContextSwitchTo(work_mcxt);
 			/* Only one chunk should match */
 			Assert(ts_scan_iterator_next(&chunk_it) == NULL);
```

Comment:
I know that this is not related to this PR, but I think placing working code inside the `Assert` statement might break it for the Release build or any other one, which builds without debug enabled, etc

Reply:
It verifies that only one row is returned, which is the assumption we make since we scan on the primary key. In debug, this is verified to be true, but the check isn't really needed if the assumption is true, which is why it is elided in release builds.